src/supervisor/message_handlers.py

```diff
@@ -66,19 +66,22 @@ def send_message(self, message_details):
             have .log_file and .attachments
         """
 
-        log = logging.getLogger(message_details.project_name)
-
         #: Configure outgoing settings
-        from_address = self.email_settings['from_address']
-        to_addresses = self.email_settings['to_addresses']
-        smtp_server = self.email_settings['smtpServer']
-        smtp_port = self.email_settings['smtpPort']
-
-        if None in [from_address, to_addresses, smtp_server, smtp_port]:
-            log.warning('Required environment variables for sending emails do not exist. No emails sent.')
-
+        try:
+            from_address = self.email_settings['from_address']
+            to_addresses = self.email_settings['to_addresses']
+            smtp_server = self.email_settings['smtpServer']
+            smtp_port = self.email_settings['smtpPort']
+
+        except KeyError:
+            warnings.warn('Required environment variables for sending emails do not exist. No emails sent.')
```

Comment:
**nitpick**: this isn't an environment variable, it's a constructor property.

Reply:
That's what I get for cribbing from forklift. I mean, technically, they're keys in a dictionary that itself is a constructor property :)

I'm wracking my brain, but the best substitute I can come up with is "Required email settings don't exist. No emails sent" and "Required email settings exist but aren't populated. No emails sent." Thoughts?

Reply:
yeah forklift grabs those values from os.environ if i remember correctly. but, in this case, your substitutions seem like nice improvements.

Reply:
I think I copied the email_settings dict straight from the forklift config code, but I could be wrong. I'll change the wording accordingly.

Reply:
I guess ideally I'd design an object that checks all this at creation time so send_message() can rely on the values being there and populated.

Reply:
yes unless for some reason you'd want to create the object and fill out any details later but that might be contrived.

Reply:
Logged as #6 .